Update release documentation about release_helper.py script (#10502)
ComplianceAsCode/content: docs/release_process.md

How to Perform An Upstream Release (GitHub)
===========================================

# Automation of Release Tasks

> In line with this release process documentation, there is a script designed to automate as much as possible the tasks described in this document. While any task can be performed manually, it is highly recommended to use the `release_helper.py` script during the release process in order to increase the efficiency and be less prone to errors.

Reviewer: ```suggestion
In line with this release process documentation, there is a script designed to automate the tasks described in this document as much as possible. While any task can be performed manually, it is highly recommended to use the `release_helper.py` script during the release process in order to increase the efficiency and be less prone to errors.
```

> In line with this release process documentation, there is a script designed to automate as much as possible the tasks described in this document. While any task can be performed manually, it is highly recommended to use the `release_helper.py` script during the release process in order to increase the efficiency and be less prone to errors.

Author: Thanks for reminding me about this. I will fix.

> The script interacts with the Github API so an api token is necessary for most of the tasks. Instructions for generating a Github API token can be found directly on [Github documentation](https://docs.github.com/) but some hints are also provided by the script.

Reviewer 2: ```suggestion

```
This seems to be a duplicate of the line below.

The script interacts with the Github API, so an [API token](https://github.com/settings/tokens) is required for most tasks. Instructions for generating a Github API token can be found directly in the [Github documentation](https://docs.github.com/), but some hints are also provided by the script.

The `release_helper.py` script can also be used to check the current status of the project regarding releases:

```bash
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content stats
```

> Throughout the document, for each automated task it will be shown an example of the respective task using the script but here is a shortcut for quicker reference:

Reviewer: ```suggestion
There is an example usage of release_helper.py  shown for each automated task throughout the document. But here is a cheatsheet for quicker reference:
```

```bash
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release_prep --contributors
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release_prep --branch
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release_prep --milestone
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release_prep --issues
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release_prep --bump-version
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release --message
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content release --tag
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content cleanup --branch
./release_helper.py -c ~/secret.ini -r ComplianceAsCode/content finish --message
```

## How to Contribute

This script was created targeting automation of boring tasks and reduction of human errors while allowing release process to be executed much faster with a lower knowledge curve.

> There are still space for improvements and any contribution is welcome. Here are some ideas to start contributing:

Reviewer: ```suggestion
There is still space for improvement and any contribution is welcome. Here are some ideas to start contributing:
```

> * Include an option to automate the process of updating the **stable** branch

Reviewer: As I am looking at those ideas, wouldn't it be better to create them as upstream issues? I am not sure that they will be discovered here.

Author: It is also a good idea to create them as upstream issues. Some improvements are relatively simple. I will review them soon (before the next release). After this review, if something remains in this list or if new ideas appear, I can file upstream issues for them.
